apps/server/src/modules/user/legacy/uc/admin-api-users.uc.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@nestjs/common';
+import { User } from '@shared/domain/entity';
+import { Permission, RoleName } from '@shared/domain/interface';
+import { UserRepo } from '@shared/repo';
+import { ForbiddenOperationError } from '@shared/common';
+import { RoleService } from '../../../role';
+import { AuthorizationService } from '../../../authorization';
+import { UsersAdminContextEnum } from '../enum';
+import { UserByIdParams, UserListResponse, UserResponse, UsersSearchQueryParams } from '../controller/dto';
+import { AdminUsersService } from '../service';
+
+@Injectable()
+export class AdminApiUsersUc {
+	constructor(
+		private readonly userRepo: UserRepo,
+		private readonly roleService: RoleService,
+		private readonly adminUsersService: AdminUsersService,
+		private readonly authorizationService: AuthorizationService
+	) {}
+
+	public async findUsersByParams(
+		context: UsersAdminContextEnum,
+		currentUserId: string,
+		params: UsersSearchQueryParams
+	): Promise<UserListResponse> {
+		const currentUser = await this.userRepo.findById(currentUserId, true);
+		this.validateAccessToContext(context, currentUser);
+		const { school } = currentUser;
+		const currentSchoolYear = school.currentYear;
+		const currentSchoolYearId = currentSchoolYear?.id;
+		const contextRole = await this.getRoleForContext(context);
+
+		return this.adminUsersService.getUsersWithNestedData(contextRole?.id, school.id, currentSchoolYearId, params);
+	}
+
+	public async findUserById(
+		context: UsersAdminContextEnum,
+		currentUserId: string,
+		params: UserByIdParams
+	): Promise<UserResponse> {
+		const currentUser = await this.userRepo.findById(currentUserId, true);
+		this.validateAccessToContext(context, currentUser);
+		const { school } = currentUser;
+		const currentSchoolYear = school.currentYear;
+		const currentSchoolYearId = currentSchoolYear?.id;
+		const contextRole = await this.getRoleForContext(context);
+
+		return this.adminUsersService.getUserWithNestedData(contextRole?.id, school.id, currentSchoolYearId, params.id);
+	}
+
+	private validateAccessToContext(context: UsersAdminContextEnum, currentUser: User) {
+		const permission = this.getPermissionForContext(context);
+		try {
+			this.authorizationService.checkAllPermissions(currentUser, [permission]);
+		} catch (e) {
+			throw new ForbiddenOperationError(`Current user is not authorized to search for ${context.valueOf()}.`);
```

Comment:
Why do you map the error here? Is it to fix the wrong error in the underlying method?

Reply:
Yeah, exactly, we should use ForbiddenLoggableException instead of UnauthorizaedException

Reply:
Okay, but shouldn't you use `ForbiddenLoggableException` here then? And if so, please add a comment that this is only a temporary fix for the problem in the underlying method.